### PR TITLE
feat: Expose paint from svgComponent to set opacity and have opacity effects

### LINKED ISSUE #2091: set opacity on SvgComponent
# Problem to solve

currently you can't set opacity on `SvgComponent` as paint in svg is private.

# Proposal

expose paint from svg to so can get paint and set opacity on `SvgComponent`

# More information

yes i am intersted on working on this

### PR DESCRIPTION
**Pull request by dipakp2726**
# Description
currently you can't set opacity on svgComponent as its paint is private.
exposed paint from svgComponent Using HasPaint Mixin to so can get paint and set opacity and have opacity effects

## usage
```
 final android = SvgComponent(
      svg: svgInstance,
      position: Vector2.all(100),
      size: Vector2.all(100),
    );

 android.paint.color =   android.paint.color.withOpacity(.6);
```

## Checklist
- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have followed the [Contributor Guide] when preparing my PR.
- [ ] I have updated/added tests for ALL new/updated/fixed functionality.
- [ ] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [ ] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?
- [x] No, this PR is not a breaking change.

## Related Issues
closes #2091 set opacity on SvgComponent

### CODE REVIEW

**Comment by ufrshubham** [comment]
I think a better idea would be to make SvgComponent use `HasPaint` so that it works with OpacityEffect as well.

**Comment by dipakp2726** [comment]
ok i will  add it with has component

**Comment by dipakp2726** [comment]
exposed paint With `HasPaint` mixin

File: packages/flame_svg/lib/svg_component.dart [hunk 1]
```diff
@@ -18,7 +18,14 @@ class SvgComponent extends PositionComponent {
     super.anchor,
     super.children,
     super.priority,
-  }) : _svg = svg;
+    Paint? paint,
+  }) {
+    _svg = svg;
+
```
Comment (spydon) [inline packages/flame_svg/lib/svg_component.dart:24]:
```suggestion
  }) : _svg = svg {
```

File: packages/flame_svg/lib/svg_component.dart [hunk 2]
```diff
@@ -18,7 +18,12 @@ class SvgComponent extends PositionComponent {
     super.anchor,
     super.children,
     super.priority,
-  }) : _svg = svg;
+    Paint? paint,
+  }) : _svg = svg {
+    if (paint != null) {
+      this.paint = paint;
+    }
```
Comment (ufrshubham) [inline packages/flame_svg/lib/svg_component.dart:25]:
```suggestion
    this.paint = paint ?? this.paint..filterQuality = FilterQuality.high;
```

File: packages/flame_svg/lib/svg_component.dart [hunk 3]
```diff
@@ -30,7 +35,7 @@ class SvgComponent extends PositionComponent {
 
   @override
   void render(Canvas canvas) {
-    _svg?.render(canvas, size);
+    _svg?.render(canvas, size, overridePaint: paint);
```
Comment (ufrshubham) [inline packages/flame_svg/lib/svg_component.dart:38]: Lets make it a non-optional parameter and remove `_paint` from `Svg` altogether

**Review by ufrshubham — COMMENTED** [review]
Since `SvgComponent` now has a paint, we can get rid of `_paint` from `Svg`

File: packages/flame_svg/lib/svg_component.dart (hunk 3 above)
Comment (dipakp2726) [reply packages/flame_svg/lib/svg_component.dart]: done

Comment (spydon) [reply packages/flame_svg/lib/svg_component.dart]: We shouldn't, `Svg` can be used outside of FCS.

Comment (spydon) [reply packages/flame_svg/lib/svg_component.dart]: Sorry @dipakp2726, the paint needs to exist on both just like it was. 😅

Comment (dipakp2726) [reply packages/flame_svg/lib/svg_component.dart:37]: reverted.

File: packages/flame_svg/lib/svg.dart [hunk 4]
```diff
@@ -21,6 +21,7 @@ class Svg {
 
   final MemoryCache<Size, Image> _imageCache = MemoryCache();
 
+  /// paint on svg
```
Comment (spydon) [inline packages/flame_svg/lib/svg.dart:24]: This dartdoc doesn't give anything
```suggestion
```

File: packages/flame_svg/lib/svg.dart [hunk 5]
```diff
@@ -41,14 +42,21 @@ class Svg {
...
     final image = _getImage(_size);
 
     if (image != null) {
       canvas.save();
       canvas.scale(1 / pixelRatio);
-      canvas.drawImage(image, Offset.zero, _paint);
+
+      final drawPaint = overridePaint ?? _paint;
+
+      canvas.drawImage(image, Offset.zero, drawPaint);
```
Comment (spydon) [inline packages/flame_svg/lib/svg.dart:59]:
```suggestion
      final drawPaint = overridePaint ?? _paint;
      canvas.drawImage(image, Offset.zero, drawPaint);
```

File: packages/flame_svg/lib/svg_component.dart [hunk 6]
```diff
@@ -18,7 +18,11 @@ class SvgComponent extends PositionComponent {
     super.anchor,
     super.children,
     super.priority,
-  }) : _svg = svg;
+    Paint? paint,
+  }) : _svg = svg {
+    this.paint = paint ?? this.paint
+      ..filterQuality = FilterQuality.high;
```
Comment (spydon) [inline packages/flame_svg/lib/svg_component.dart:24]: Why are you setting the filterquality here? `FilterQuality.high` will make most SVGs worse.

Comment (dipakp2726) [reply packages/flame_svg/lib/svg_component.dart:24]: does it need to be removed ?

**Review by dipakp2726 — COMMENTED** [review]
took default value of paint from here

File: packages/flame_svg/lib/svg_component.dart (hunk 6 above)
Comment (spydon) [reply packages/flame_svg/lib/svg_component.dart:24]: Yeah, remove it. I guess we should remove it from the `Svg` class too, but that can be done later.

Comment (dipakp2726) [reply packages/flame_svg/lib/svg_component.dart]: removed

**Review by spydon — APPROVED** [review]

**Pull request merged**